src/GlobalDataStore.lua

```diff
@@ -113,4 +113,25 @@ function GlobalDataStore:UpdateAsync(key: string, transform)
 	return copyDeep(transformed), self.keyInfos[key]
 end
 
+function GlobalDataStore:RemoveAsync(key: string)
+	validateString("key", key, Constants.MAX_KEY_LENGTH)
+
+	if self.errors ~= nil then
+		self.errors:simulateError("RemoveAsync")
+	end
+
+	self.budget:yieldForBudget({ Enum.DataStoreRequestType.SetIncrementAsync })
+	self.yield:yield()
+
+	local oldValue = self.data[key]
+	if oldValue == nil then
+		return nil, nil
+	end
+	local keyInfo = self.keyInfos[key]
+
+	self:write(key, nil)
```

Comment:
I think this throws an error because `userIds` is nil. I checked the behavior and the `DataStoreKeyInfo` should not be removed with `RemoveAsync` however, the user ids and metadata are removed.

Reply:
fixed in 3rd commit